### DISCUSSION TITLE
TagsInput display Issue

### DISCUSSION THREAD

**Discussion by liza-nt** [filamentphp/filament#6951] [General]
I have a form that includes a select field allowing users to choose a field type. Depending on the selected field type, the form should dynamically generate different types of fields. The code works correctly for fields like TextInput, Select, and TextArea. However, I'm encountering an issue when trying to render the TagsInput field.

When I choose the TagsInput field type from the select field, the expected TagsInput field is not being rendered on the form. The issue seems to be specific to rendering the TagsInput component.
here is my code
```
   public static function getFieldOptionsSchema(Closure $get)
    {
        $fieldKey = $get('field_key');

        if ($fieldKey === 'select') {
            return [
                Select::make('field_options'),
            ];
        } if ($fieldKey === 'text') {
            return [
                TextInput::make('field_options'),
            ];
        } if ($fieldKey === 'tag') {
            return [
                TagsInput::make('field_options'),
            ];
        }

        return [];
    }
   

     Grid::make(2)->schema(function (Closure $get) use ($form) {
                    $fieldKey = Select::make('field_key')->options([
                        'tag' => 'Tag',
                        'select' => 'Select',
                        'text' => 'Text',
                    ])->reactive();

                    $fieldOptions = self::getFieldOptionsSchema($get);

                    return array_merge([$fieldKey], $fieldOptions);
                }),
                ```

**Comment by danharrin** [comment]
When the select changes, you need to loop through each dynamic field and initialise it with `$set('x', null)` or `$set('x', [])` depending on the field. For tags input, it is `[]`.

**Reply by liza-nt** [reply]
sorry i did not understand Can you explain bit more on that

**Reply by danharrin** [reply]
Where is the code for your select

**Reply by liza-nt** [reply]
inside grid schema
```

     Grid::make(2)->schema(function (Closure $get) use ($form) {
                    $fieldKey = Select::make('field_key')->options([
                        'tag' => 'Tag',
                        'select' => 'Select',
                        'text' => 'Text',
                    ])->reactive();

                    $fieldOptions = self::getFieldOptionsSchema($get);

                    return array_merge([$fieldKey], $fieldOptions);
                }),
```

**Reply by danharrin** [reply]
`->afterStateUpdated(function ($set, $state) {})` on the select

Use `$state` to work out which dynamic field is there, and then call `$set('x', x)`

**Reply by liza-nt** [reply]
Yes, it worked! Thank you very much.
I have one question though If the other fields are functioning correctly, why is the TagsInput field not being rendered?

**Reply by danharrin** [reply]
Because by default, fields have a null state, but tags input expects an array (for multiple tags), so it breaks when it isnt initialised
